lib/deprecation.nix

```diff
@@ -46,6 +46,21 @@ rec {
         '';
     };
 
+  processRenameOptions =
```

Comment:
You mentioned before that I'm this may not be the best name. Do you have any ideas we can discuss for a better name?

Maybe something that implies it's meant for use renaming to `settings` options? Or that it converts to snake case?

Perhaps `mkSettingsRenameOptionModules`? Or something along those lines?

Reply:
I changed it to that right now while doing the other clean up. Didn't know if the name should be more abstract since it's technically not limited to the settings.

Reply:
> Didn't know if the name should be more abstract since it's technically not limited to the settings.

Yeah I agree, my proposal isn't ideal because it can be more general. But in practice where else will we do this and also want to convert to snake_case?

> `mkPrefixedRenamedOptionModulesWithSnakeCase` seems more accurate but kinda wordy..

Yep, lol

Reply:
> Yeah I agree, my proposal isn't ideal because it can be more general. But in practice where else will we do this and also want to convert to snake_case?
> 
I'm fine with this way, unless someone can come up with something better and we could rename it since it's only got a few usages.